src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -150,6 +152,34 @@ impl ManagedBarrierState {
         self.create_mview_progress.clear();
     }
 
+    /// Notify unexpected actor exit with given `actor_id`.
+    pub(crate) fn notify_exit(&mut self, actor_id: ActorId) {
+        for (_, barrier_state) in &mut self.epoch_barrier_state_map {
+            match barrier_state.inner {
+                ManagedBarrierStateInner::Issued {
+                    ref remaining_actors,
+                    ref mut collect_notifier,
+                } => {
+                    if remaining_actors.contains(&actor_id) {
+                        let collect_notifier = std::mem::take(collect_notifier);
+                        if collect_notifier
+                            .unwrap()
```

Comment:
Will this `unwrap` panic if multiple actors throw error?

Reply:
Good catch, it should do nothing if the collect notifier is None.